src/components/ProfileTabbedView.tsx

```diff
@@ -0,0 +1,68 @@
+import { Tab, TabList, TabPanel, Tabs } from "react-tabs";
+import {useState} from "react";
+import {api} from "~/utils/api";
+import {useRouter} from "next/router";
+import {ProfileCard} from "./ProfileCard";
+import { Spinner } from "./Spinner";
+import {BlogPostViewer} from "./BlogPostViewer";
+import {Layout} from "./Layout";
+
+export const ProfileTabbedView: React.FC = () => {
+    const [_, setForceRerender] = useState(false);
+    const forceRerender = () => setForceRerender(!_);
+    const [tabState, setTabState] = useState(0);
+
+    const router = useRouter();
+    const { id } = router.query;
+    if(!router.isReady) {
+        return (<Spinner size={2} />);
+    }
```

Comment:
I think this is redundant. You already have a check for it in the return statement 

```jsx
    <>
      {router.isReady ?
        ...
        :
            <Spinner size={24} />}
     </>
```

Reply:
This might also cause weird race condition problems if the same check is in multiple places. In general I'd recommend keeping all the logic related to rendering jsx in the `return` statement of the component.

Reply:
after refactoring, `ProfileTabbedViewer` isn't responsible for this anymore